app/controllers/graphs_controller.rb

```diff
@@ -15,15 +15,12 @@ def reports_by_hour
   end
 
   def reason_counts
-    @posts = Post.where('created_at > ?', params[:months].to_i.months.ago || 3.months.ago)
-    @posts = @posts.where(site_id: params[:site_id]) if params[:site_id].present?
-    # render json: Reason.all.map { |r| [r.reason_name, @posts.group(:reasons).count]}
-    puts "Start it"
-    reasons = Reason.all.includes(:posts)
-    puts "Reasons gotten"
-    @posts.load
-    render json: reasons.map { |r| [r.reason_name, (r.posts & @posts).count] }.sort_by { |name, count| count }
-    puts "End it"
+    render json: Reason.joins(:posts)
+      .where('posts.created_at >= ?', params[:months].to_i.months.ago || 3.months.ago)
+      .where(params[:site_id].present? ? { posts: { site_id: 1 } } : {})
+                       .group(:reason_name)
+      .count
```

Comment:
Weird indentation... rubocop will probably fix it for you if you tell it to

Reply:
Weirdly enough, rubocop fixed it to become this, then fixed it again later.